### PR TITLE
Refractor TensorBox Code

### PR DESCRIPTION
**Pull request by MarvinTeichmann**
This pull request retractors the training code. It makes the whole code pep8 compatibly and splits the model building in meaningful parts. The refactoring make the code much more readable and understandable for outsiders.

The model building process is split into the following functions:
1. data_input.inputs
2. encoder.inference
3. objective.decoder
4. objective.loss
5. optimizer.training
6. objective.evaluation

Additionally, the code is now split into different files. This has the advantage, that it is easier to test new models. One could for example manipulate the encoder (use ResNet) or the loss/decoder while keeping the other parts constant.

### CODE REVIEW

**Comment by MarvinTeichmann** [comment]
The new concept can best be seen in the `build` function of `train.py`. Take a short look at that.
